### PR TITLE
chore(dep): upgrade to puppeteer 13.1.2

### PR DESCRIPTION
**Pull request by connorjclark**

### CODE REVIEW

**Comment by connorjclark** [comment]
Screenshots used during testing can be 1+ MB, which trips up `jest-image-snapshot` running the diffing in a child process. The buffer size it uses is node's default of ~200KB. `runDiffImageToSnapshot` is what throws in `node_modules/@storybook/addon-storyshots-puppeteer/node_modules/jest-image-snapshot/src/diff-snapshot.js`.

Workaround is to use `runInProcess: true`.

ref https://github.com/americanexpress/jest-image-snapshot/issues/210

File: packages/server/test/ui/storybook.test.jsx [hunk 1]
```diff
@@ -64,9 +64,12 @@ initStoryshots({
     getMatchOptions: () => ({
       failureThreshold: process.env.CI ? 0.005 : 0.0015,
       failureThresholdType: 'percent',
+      // Slower, but required because images can be larger than the default maxBuffer for child processes
+      // which jest-image-snapshot relies on.
+      // https://github.com/americanexpress/jest-image-snapshot/issues/210
+      runInProcess: true,
     }),
     getScreenshotOptions: () => ({
-      encoding: 'base64',
```
Comment (connorjclark) [inline packages/server/test/ui/storybook.test.jsx:69]: I was getting this error:

```
Invalid file signature

      at Object.<anonymous>.module.exports.Object.<anonymous>.Parser._parseSignature (node_modules/pngjs/lib/parser.js:53:18)
      at Object.<anonymous>.module.exports.Object.<anonymous>.SyncReader.process (node_modules/pngjs/lib/sync-reader.js:34:17)
      at Object.<anonymous>.module.exports (node_modules/pngjs/lib/parser-sync.js:68:10)
      at Object.<anonymous>.exports.read (node_modules/pngjs/lib/png-sync.js:10:10)
      at diffImageToSnapshot (node_modules/@storybook/addon-storyshots-puppeteer/node_modules/jest-image-snapshot/src/diff-snapshot.js:119:39)
      at Object.toMatchImageSnapshot (node_modules/@storybook/addon-storyshots-puppeteer/node_modules/jest-image-snapshot/src/index.js:187:7)
      at __EXTERNAL_MATCHER_TRAP__ (node_modules/expect/build/index.js:342:30)
      at Object.throwingMatcher [as toMatchImageSnapshot] (node_modules/expect/build/index.js:343:15)
      at node_modules/@storybook/addon-storyshots-puppeteer/dist/imageSnapshot.js:74:43
      at step (node_modules/@storybook/addon-storyshots-puppeteer/dist/imageSnapshot.js:44:23)
      at Object.next (node_modules/@storybook/addon-storyshots-puppeteer/dist/imageSnapshot.js:25:53)
      at fulfilled (node_modules/@storybook/addon-storyshots-puppeteer/dist/imageSnapshot.js:16:58)
          at runMicrotasks (<anonymous>)
```

I walked through the stack here and it seemed the expectation is that the image data is provided to `jest-image-snapshot` as a Buffer... not sure how this was working before, but I deleted the `encoding: base64` (defaults to binary/Buffer) and the error went away.

File: packages/cli/test/test-utils.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ function cleanStdOutput(output) {
     .replace(/(\s+at \S+) .*/g, '$1')
     .replace(/\s+at (async|processTicksAndRejections|process._tickCallback)(?=\n|$)/g, '')
     .replace(
-      /appspot.com\/reports\/[0-9-]+.report.html/,
+      /appspot.com\/reports\/[0-9-]+.report.html/g,
```
Comment (connorjclark) [inline packages/cli/test/test-utils.js:63]:
```
FAIL packages/cli/test/autorun-github.test.js (51.372s)
  ● Lighthouse CI autorun CLI with GitHub status check › should submit status check to GitHub

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `Lighthouse CI autorun CLI with GitHub status check should submit status check to GitHub 1`

    - Snapshot  - 1
    + Received  + 1

    @@ -16,11 +16,11 @@


      Uploading median LHR of http://localhost:XXXX/bad.html...success!
      Open the report at storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/XXXX-XXXX.report.html
      Uploading median LHR of http://localhost:XXXX/good.html...success!
    - Open the report at storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/XXXX-XXXX.report.html
    + Open the report at storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/XXXX-113.report.html
      GitHub token found, attempting to set status...
      GitHub accepted "failure" status for "lhci/url/bad.html".
      GitHub accepted "success" status for "lhci/url/good.html".

      ↵

      70 |     );
      71 | 
    > 72 |     expect(stdout).toMatchInlineSnapshot(`
         |                    ^
      73 |       "✅  .lighthouseci/ directory writable
      74 |       ✅  Configuration file found
      75 |       ✅  Chrome installation found

      at Object.<anonymous> (packages/cli/test/autorun-github.test.js:72:20)

 › 1 snapshot failed.
```

File: package.json [hunk 3]
```diff
@@ -81,7 +80,7 @@
     "preact-render-to-json": "^3.6.6",
     "prettier": "1.16.4",
     "prop-types": "^15.7.2",
-    "puppeteer": "^1.18.1",
+    "puppeteer": "^13.0.1",
```
Comment (connorjclark) [inline package.json:83]: will do 13.1.1 if this current CI run passes in a minute

**Comment by connorjclark** [comment]
seems storybook (but only on mac??) is not including `packages/server/src/ui/routes/build-view/build-hash-selector.css` / other css files...

**Comment by connorjclark** [comment]
newest chrome resulted in tons of minute text rendering diffs. can't even distinguish them myself but apparently the pixels are different. hence, updating tons of snapshots.

**Comment by connorjclark** [comment]
RE: https://github.com/GoogleChrome/lighthouse-ci/pull/740#issuecomment-1013550392

Not sure why, but with the newest storybook this isn't an issue anymore.

File: packages/server/.storybook/preview.css [hunk 4]
```diff
@@ -4,11 +4,6 @@
  * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
  */
 
-/* html,
-body {
-  min-height: 100vh;
-} */
-
 @import 'https://fonts.googleapis.com/css?family=Roboto:400,500&display=block';
```
Comment (paulirish) [inline packages/server/.storybook/preview.css:7]: i think these lines can also be removed now (but not 100% sure)

**Review by paulirish — APPROVED** [review]
> Workaround is to use runInProcess: true

looks you managed to drop this after the storybook upgrades?  that's good news.

File: packages/server/.storybook/preview.css (hunk 4 above)
Comment (connorjclark) [reply packages/server/.storybook/preview.css:7]: No, this is the only way that storybook can have these fonts. The only other place they are added is to the app entry point, which storybook does not use.

Comment (paulirish) [reply packages/server/.storybook/preview.css:7]: Oh gotcha

**Pull request merged**